oak-core/src/main/java/org/apache/jackrabbit/oak/namepath/impl/LocalNameMapper.java

```diff
@@ -91,9 +91,10 @@ public synchronized String getJcrName(@NotNull String oakName) {
                         }
                     }
                 }
+            } else {
+                throw new IllegalStateException("No namespace mapping found for " + oakName);
```

Comment:
With that we are back at the same exception as before :-)

Reply:
actually we are now throwing an exception even earlier (i.e. for the first getName() without any mapping)

Reply:
That was an intermediate version. This current one should hopefully do, except for the problem I mentioned on Jira, which probably is something completely different.

Reply:
I wouldn't understand why `Session.addNode("foo:bar").getName()` wouldn't end up here, because

a) uri is null (there is no namespace URI registered for "foo" neither globally nor locally
b) there is no local namespace mapping for "foo"

This previously didn't lead to an exception. Now it does.